bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/editor/EditorConfigurationPage.java

```diff
@@ -0,0 +1,259 @@
+
+	private final String id = "org.eclipse.cdt.lsp.editor.preferencePage"; //$NON-NLS-1$
+
+	protected Configuration configuration;
+	protected IWorkspace workspace;
+
+	protected IWorkingCopyManager manager;
+
+	private Link link;
+	private Button specific;
+	private Control control;
+	protected ConfigurationArea area;
+
+	@Override
+	public void init(IWorkbench workbench) {
+		this.configuration = workbench.getService(Configuration.class);
+		this.workspace = workbench.getService(IWorkspace.class);
+	}
+
+	@Override
+	public void setContainer(IPreferencePageContainer container) {
+		super.setContainer(container);
+		if (manager == null) {
+			manager = Optional.ofNullable(container)//
+					.filter(IWorkbenchPreferenceContainer.class::isInstance)//
+					.map(IWorkbenchPreferenceContainer.class::cast)//
+					.map(IWorkbenchPreferenceContainer::getWorkingCopyManager)//
+					.orElseGet(WorkingCopyManager::new);
+		}
+		if (configuration == null) {
+			configuration = getConfiguration();
+		}
+		if (workspace == null) {
+			workspace = PlatformUI.getWorkbench().getService(IWorkspace.class);
+		}
+	}
+
+	protected Configuration getConfiguration() {
+		return PlatformUI.getWorkbench().getService(Configuration.class);
+	}
+
+	@Override
+	protected Label createDescriptionLabel(Composite parent) {
+		if (projectScope().isPresent()) {
+			Composite composite = new Composite(parent, SWT.NONE);
+			composite.setFont(parent.getFont());
+			composite.setLayout(GridLayoutFactory.fillDefaults().numColumns(2).create());
+			composite.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+			specific = new Button(composite, SWT.CHECK);
+			specific.setLayoutData(new GridData(SWT.BEGINNING, SWT.TOP, true, false));
+			specific.setText(LspUiMessages.LspEditorConfigurationPage_enable_project_specific);
+			specific.setFont(JFaceResources.getDialogFont());
+			specific.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> specificSelected()));
+			link = createLink(composite, LspUiMessages.LspEditorConfigurationPage_configure_ws_specific);
+			link.setLayoutData(new GridData(SWT.END, SWT.TOP, false, false));
+			Label line = new Label(composite, SWT.SEPARATOR | SWT.HORIZONTAL);
+			line.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, false, 2, 1));
+			line.setFont(composite.getFont());
+		}
+		return super.createDescriptionLabel(parent);
+	}
+
+	private void specificSelected() {
+		enableProjectSpecificSettings(specific.getSelection());
+		refreshWidgets(configuration.options(getElement()));
+	}
+
+	private Link createLink(Composite composite, String text) {
+		Link link = new Link(composite, SWT.NONE);
+		link.setFont(composite.getFont());
+		link.setText("<A>" + text + "</A>"); //$NON-NLS-1$ //$NON-NLS-2$
+		link.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				if (PreferencesUtil.createPreferenceDialogOn(getShell(), getPreferenceId(), new String[] { id }, null)
```

Comment:
I think this addresses having the node missing in the tree:

```suggestion
				if (PreferencesUtil.createPreferenceDialogOn(getShell(), getPreferenceId(), new String[] { id, getPreferenceId() }, null)
```

Reply:
Should be fixed now